tests/foreman/api/test_discoveredhost.py

```diff
@@ -55,9 +61,94 @@ def _create_discovered_host(name=None, ipaddress=None, macaddress=None):
     })
 
 
+class HostNotDiscoveredException(Exception):
+    """Raised when host is not discovered"""
+
+
 class DiscoveryTestCase(APITestCase):
     """Implements tests for foreman discovery feature"""
 
+    def _assertdiscoveredhost(self, hostname):
+        """Check if host is discovered and information about it can be
+        retrieved back
+
+        Introduced a delay of 300secs by polling every 10 secs to get expected
+        host
+        """
+        for _ in range(30):
+            discovered_host = entities.DiscoveredHost().search(
+                query={'search': 'name={}'.format(hostname)})
+            if discovered_host:
+                break
+            time.sleep(10)
```

Comment:
How about using `wait_for` here?
We pull it in as dependency and have couple of tests using it already.

Reply:
I see the existing implementation is much easier and with less efforts(LOC) than using wait_for like this :
```
        try:
            wait_for(
                lambda: entities.DiscoveredHost(user_config or default_config).search(
                    query={'search': 'name={}'.format(hostname)}),
                timeout=300,
                delay=10,
                logger=self.logger
            )
        except TimeoutError:
            raise HostNotDiscoveredException(
                "The host {} is not discovered within 300 seconds".format(hostname))
        return entities.DiscoveredHost(user_config or default_config).search(
                    query={'search': 'name={}'.format(hostname)})[0]
```

Reply:
To save 10 lines, I am ending up writing 13 lines :) with no extra benefits.

Reply:
`wait_for` returns call result, so this would be more like:
```python
try:
    res = wait_for(lambda: entities.DiscoveredHost.search(), *args)
except:
    raise HostNotDiscoveredException(...)
return res['out'][0]
```
And then, wait_for already raises exception, so unless you have to rewrite it, you could get away with:
```python
res = wait_for(lambda: entities.DiscoveredHost.search(), *args)
return res['out'][0]
```

Of course it's your choice to make, that was only a suggestion.

Reply:
Just observed, It's not ```TimeoutError``` its ```TimedoutError``` for which the 'wait_for' doesnt provides an exception to be handled. And I dont want to miss any other bug by simply doing except :(
Complexed filling.

Keeping as it is for now @mirekdlugosz , let me know if you wanna go hard on it. Thanks for the comments though.

Reply:
No, use whichever way you prefer. This PR is ACK from me already, with or without `wait_for`.